src/Jackalope/Transport/DoctrineDBAL/Query/QOMWalker.php

```diff
@@ -294,16 +294,25 @@ public function walkSelectorSource(QOM\SelectorInterface $source)
      *
      * @throws NotImplementedException
```

Comment:
ah, then we should add that, saying `@throws NotImplementedException if the right side of a join is a nested join`